core/src/main/scala/org/bitcoins/core/hd/BIP32Path.scala

```diff
@@ -6,6 +6,76 @@ import org.bitcoins.core.number.UInt32
 abstract class BIP32Path {
   def path: Vector[BIP32Node]
 
+  /**
+    * BIP32 paths can be subsets/superset of each other.
+    * If all elements in a path `p` is included in a path
+    * `P`, (i.e. `p` is a subset of `P`), `p.diff(P)`
+    * is the elements from `P` that is not in `p`.
+    *
+    * @example
+    * {{{
+    *  // equal paths
+    * m/44'/1' diff m/44'/1' == Some(BIP32Path.empty)
```

Comment:
Maybe should return `None` here?

Reply:
`m/44'/1' derive BIP32Path.empty` is a completely valid operation though
Any path is a subset of itself, and if diffing is defined for subsets, then diffing something with itself should also be defined

Not that big of a point though, I'm fine with both. Do you have strong feelings on it?